lib/mcstat.inc

```diff
@@ -98,47 +98,40 @@ function hr_time($seconds) {
  */
 class mcstat_client
 {
-  private $socket_ptr = null;
+  private $mc = null;
   private $hostname = null;
   private $port = null;
+  private $username = null;
+  private $password = null;
 
-  function __construct($hostname = DEFAULT_MEMCACHE_HOST, $port = DEFAULT_MEMCACHE_PORT) {
+  function __construct($hostname, $port, $username, $password) {
+    $this->mc = new Memcached();
```

Comment:
Is it possible to add a check somewhere to ensure that the correct PHP extension is installed. IIRC there are two memcache-related PHP extensions.

Reply:
Yep, there is [Memcache](https://www.php.net/manual/en/class.memcache.php) and [Memcached](https://www.php.net/manual/en/class.memcached.php). I'll look into adding a check to make sure the extension is installed.

Reply:
Added check at the top of this constructor to throw an Exception if the proper extension is not loaded.